### PR TITLE
Customer records

### PR DESCRIPTION
**Pull request by wlffann**
@case-eee @drod1000 
This includes all the customers endpoints.

Some merchant tests are failing, but merging the merchant-records branch should solve these problems.

### CODE REVIEW

File: app/controllers/api/v1/search_customers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    render json: Customer.where(customers_params)
+  end
+
+  def show
+    render json: Customer.find_by(customers_params)
+  end
+
+  private
+
+  def customers_params
```
Comment (case-eee) [inline app/controllers/api/v1/search_customers_controller.rb:14]: 👍

File: app/controllers/api/v1/search_merchants_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class Api::V1::SearchMerchantsController < ApplicationController
```
Comment (case-eee) [inline app/controllers/api/v1/search_merchants_controller.rb:1]: could we use the strong params here as well?

File: db/migrate/20170124154026_change_names_to_ci_customers.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class ChangeNamesToCiCustomers < ActiveRecord::Migration[5.0]
+  def change
+    enable_extension :citext
```
Comment (case-eee) [inline db/migrate/20170124154026_change_names_to_ci_customers.rb:3]: 👍

**Review by case-eee — COMMENTED** [review]
looking good 🎉

**Comment by wlffann** [comment]
This branch is now merged with the updated merchants and is ready to merge into master!

**Comment by drod1000** [comment]
Looks good! merging

**Pull request merged**